File: Codesnippet/Codesnippet_Go/CodeSnippet_test.go
package checkout

import (
	"fmt"
	"github.com/adyen/adyen-go-api-library/v9/src/adyen"
	"github.com/adyen/adyen-go-api-library/v9/src/checkout"
	"github.com/adyen/adyen-go-api-library/v9/src/common"
	"golang.org/x/net/context"
	"testing"
)

func Test_Snippet_Session(t *testing.T) {

	t.Run("Session", func(t *testing.T) {

		client := adyen.NewClient(&common.Config{
			ApiKey:"YOUR_API_KEY",
			Environment: common.TestEnv,
		})
		service := client.Checkout()

		req := service.PaymentsApi.SessionsInput()
		req = req.CreateCheckoutSessionRequest(checkout.CreateCheckoutSessionRequest{
			MerchantAccount: "YOUR_MERCHANT_ACCOUNT",
			Amount: checkout.Amount{
				Value:    1250,
				Currency: "EUR",
			},
			Reference:   "ref",
			CountryCode: common.PtrString("NL"),
			ReturnUrl:   "https://your-company.com/checkout?shopperOrder=12xy...",
		})
		res, _, _ := service.PaymentsApi.Sessions(context.Background(), req)
		fmt.Print(res.ReturnUrl)//prints
	})
}

File: Codesnippet_node/CodeSnippet.test.ts
//test imports
import nock from "nock";
import { SessionResultResponse } from "../typings/checkout/sessionResultResponse";

// import the required class
import {CheckoutAPI} from "../services";
import Client from "../client";
import { checkout } from "../typings";

//setup client and service
const client = new Client({ apiKey: "YOUR_API_KEY", environment: "TEST" });
const checkoutApi = new CheckoutAPI(client);
// const requestOptions = { idempotencyKey: "YOUR_IDEMPOTENCY_KEY" };???

//create session request
checkoutApi.PaymentsApi.sessions({
    amount: { currency: "EUR", value: 1000 },
    reference: "YOUR_PAYMENT_REFERENCE",
    returnUrl: "https://your-company.com/checkout?shopperOrder=12xy..",
    merchantAccount: 'YOUR_MERCHANT_ACCOUNT',
    countryCode: "NL",
    channel: checkout.PaymentSetupRequest.ChannelEnum.Web,
})
    .then((response) => {
        console.log(response);
    })
    .catch((e) => {
        console.log(e);
    })

// let client: Client;
let checkoutService: CheckoutAPI;
let scope: nock.Scope;

beforeEach((): void => {
    if (!nock.isActive()) {
        nock.activate();
    }
    // client = createClient();
    scope = nock("https://checkout-test.adyen.com/v71");
    checkoutService = new CheckoutAPI(client);
});

afterEach(() => {
    nock.cleanAll();
});

    test("Should create basic payment", async(): Promise<void> => 
    {
        scope.get("/sessions/mySessionIdMock?sessionResult=sessionResult")
        .reply(200, {
            "id": "CS12345678",
            "status": "completed"
        });

        const resultOfPaymentSessionResponse = await checkoutService.PaymentsApi.getResultOfPaymentSession("mySessionIdMock", "sessionResult");
        expect(resultOfPaymentSessionResponse.id).toEqual("CS12345678");
        expect(resultOfPaymentSessionResponse.status).toEqual(SessionResultResponse.StatusEnum.Completed);
    });
